### DISCUSSION TITLE
How to avoid fetch warnings when called from component

### DISCUSSION THREAD

**Discussion by akvadrako** [sveltejs/kit#8627] [General]
I have some components that call fetch on the client side and I can't find a way to avoid warnings like this:

> Loading XXX using `window.fetch`. For best results, use the `fetch` that is passed to your `load` function: https://kit.svelte.dev/docs/load#making-fetch-requests

I have tried placing the fetch in `onMount()`, `setTimeout()` and `queueMicrotask()` handlers, but none of that helps. Is there some way to access the `native_fetch` function directly?

This is on Firefox 109 and @sveltejs/kit 1.2.0.

**Comment by teemingc** [comment]
I wasn't able to reproduce this. Could you help provide a reproduction?
The warning should only be displayed when using `fetch` inside a `load` function running in the browser.

**Reply by akvadrako** [reply]
Sure, here is a minimal demo: https://stackblitz.com/edit/sv-bug-fetch-demo

I am calling fetch from within onMount().

**Reply by teemingc** [reply]
Weirdly enough I can't seem to find the warning in the browser console from that demo.

**Reply by akvadrako** [reply]
Are you using a recent Firefox version?

**Reply by akvadrako** [reply]
If this is unexpected behavior, I can make an issue. I thought it was user error.

**Reply by geoffrich** [reply]
This should've been fixed with https://github.com/sveltejs/kit/pull/8456 in v1.0.12, but I do still see the warning in Firefox with the repro. I'll re-open the issue.

**Reply by geoffrich** [reply]
See https://github.com/sveltejs/kit/issues/7992 for the reopened issue.

**Reply by teemingc** [reply]
> Are you using a recent Firefox version?

I was using Chrome. I see now that the issue is with Firefox as Geoff mentioned.
